packages/hub/src/lib/parse-safetensors-metadata.ts

```diff
@@ -223,27 +252,45 @@ export async function parseSafetensorsMetadata(
 		throw new TypeError("Only model repos should contain safetensors files.");
 	}
 
-	if (RE_SAFETENSORS_FILE.test(params.path ?? "") || (await fileExists({ ...params, path: SAFETENSORS_FILE }))) {
+	if (
+		(params.path && RE_SAFETENSORS_FILE.test(params.path)) ||
+		(await fileExists({ ...params, path: SAFETENSORS_FILE }))
+	) {
 		const header = await parseSingleFile(params.path ?? SAFETENSORS_FILE, params);
 		return {
 			sharded: false,
 			header,
-			...(params.computeParametersCount && {
-				parameterCount: computeNumOfParamsByDtypeSingleFile(header),
-			}),
+			...(params.computeParametersCount
+				? {
+						parameterCount: computeNumOfParamsByDtypeSingleFile(header),
+						parameterTotal:
+							/// shortcut: get param count directly from metadata
+							header.__metadata__.total_parameters
+								? parseInt(header.__metadata__.total_parameters.toString())
+								: undefined,
+				  }
+				: undefined),
 		};
 	} else if (
-		RE_SAFETENSORS_INDEX_FILE.test(params.path ?? "") ||
+		(params.path && RE_SAFETENSORS_INDEX_FILE.test(params.path)) ||
 		(await fileExists({ ...params, path: SAFETENSORS_INDEX_FILE }))
 	) {
-		const { index, headers } = await parseShardedIndex(params.path ?? SAFETENSORS_INDEX_FILE, params);
+		const path = params.path ?? SAFETENSORS_INDEX_FILE;
+		const index = await parseShardedIndex(path, params);
+		const shardedMap = await fetchAllHeaders(path, index, params);
+
 		return {
 			sharded: true,
 			index,
-			headers,
-			...(params.computeParametersCount && {
-				parameterCount: computeNumOfParamsByDtypeSharded(headers),
-			}),
+			headers: shardedMap,
+			...(params.computeParametersCount
+				? {
```

Comment:
The parameter counting logic is mixed with metadata parsing (i.e: total_parameters), I feel it should have a clear separation between metadata parsing and parameter computation

Reply:
in practice we always want to compute parameters, so I think it's fine